include/base/NekRSProblem.h

```diff
@@ -106,7 +106,7 @@ class NekRSProblem : public NekRSProblemBase
    */
   virtual double minInterpolatedTemperature() const;
 
-  virtual bool movingMesh() const override { return _moving_mesh; }
+  virtual const bool hasMovingNekMesh() const override { return _nek_mesh->movingMesh(); }
```

Comment:
Is there a reason why this implementation of `hasMovingNekMesh()` couldn't be placed into the `NekRSProblemBase`, and shared by all derived classes?

Reply:
I actually wasn't sure if it needed to be in the base class, but I'm happy to change that. I think the previous version of this in `NekRSProblemBase` just returned `false`??

Reply:
Oh, you're right. Ok I think I recall why we had `moving_mesh` be on `NekRSProblem`, since it only made sense for that class. With these changes, we should technically be throwing a warning if someone sets `moving_mesh` on `NekRSMesh` but pairs it with `NekRSStandaloneProblem`/`NekRSSeparateDomainProblem` (because it's unused).

Which do you prefer?

- Keep `moving_mesh` on `NekRSProblem`, so there's no ambiguity about when it is used, but some things related to the mesh aren't located (most logically) on a Cardinal class that deals with meshes
- Move `moving_mesh` to `NekRSMesh`, and add extra warning checks that no one specifies `moving_mesh = true` paired with one of the NekRS-wrapper classes that don't make sense for it to be specified.

Reply:
I can add the checks to `NekRSMesh`

Reply:
Great, thanks! Yes let's add checks to NekRSMesh that throw a warning to the user if they set `moving_mesh` for anything except `NekRSProblem`.